module/VuFind/src/VuFind/Search/Solr/InjectHighlightingListener.php

```diff
@@ -118,11 +127,14 @@ public function onSearchPre(EventInterface $event)
             if ($params = $command->getSearchParameters()) {
                 // Set highlighting parameters unless explicitly disabled:
                 $hl = $params->get('hl');
-                if (!isset($hl[0]) || $hl[0] != 'false') {
+                if (($hl[0] ?? 'true') != 'false') {
                     $this->active = true;
                     $params->set('hl', 'true');
                     $params->set('hl.simple.pre', '{{{{START_HILITE}}}}');
                     $params->set('hl.simple.post', '{{{{END_HILITE}}}}');
+                    foreach ($this->extraHighlightingParameters as $key => $val) {
```

Comment:
Should the extra params be set first so that even if someone sets e.g. `hl.simple.pre` in them, it doesn't break things?

Reply:
Good idea -- adjusted accordingly!